furi/core/event_loop.c

```diff
@@ -186,13 +454,19 @@ void furi_event_loop_run(FuriEventLoop* instance) {
                         }
                     }
                 }
+
+                furi_event_loop_restore_flags(instance, flags & ~FuriEventLoopFlagEvent);
```

Comment:
why?

Reply:
The flags are being cleared upon each call to `xTaskNotifyWaitIndexed()`, so in case there is a pending `FuriEventLoopFlagRequest` together with a `FuriEventLoopFlagEvent`, the former will not get processed, so we need to restore all flags except `FuriEventLoopFlagEvent`. 

This is done to ensure that event processing gets higer priority than requests (timers, pending callbacks etc), i.e. they will not receive control until there are no more events to process.

If this is not a desirable behaviour, I can change it to a sequential one - everything will be processed in one fell swoop.